### PR TITLE
Changed integer based indexing to location based index

### PR DESCRIPTION
**Pull request by alishobeiri**
Now selects first element in series rather than 0th index item - old way could cause problems when the dataframe doesn't start at index 0 but rather starts at a higher number. Happened to me :P

### CODE REVIEW

File: plotly/figure_factory/_facet_grid.py [hunk 1]
```diff
@@ -783,7 +783,7 @@ def create_facet_grid(df, x, y, facet_row=None, facet_col=None,
                     )
     show_legend = False
     if color_name:
-        if isinstance(df[color_name][0], str) or color_is_cat:
+        if isinstance(df[color_name].iloc[0], str) or color_is_cat:
```
Comment (chriddyp) [inline plotly/figure_factory/_facet_grid.py]: are there other places in the code where this happens?

**Comment by chriddyp** [comment]
Looks good! @alishobeiri - could you add a simple test that breaks before this code was applied but is fixed after this code was applied?
@Kully - this is your section of the code, I'll let you review it more thoroughly.

File: plotly/figure_factory/_facet_grid.py (hunk 1 above)
Comment (Kully) [inline plotly/figure_factory/_facet_grid.py]: @alishobeiri There's a `elif isinstance(df[color_name][0], Number):` line a little further down the code that you can change too.

Thanks for doing this! Awesome stuff.

**Comment by alishobeiri** [comment]
```
| Unnamed: 0	carat	cut	color	clarity	depth	table	price	x	y	z	cut-labels	color-labels	clarity-labels |
|0	1	0.23	5	2	2	61.5	55.0	326	3.95	3.98	2.43	Ideal	E	SI2 |
|1	2	0.21	4	2	3	59.8	61.0	326	3.89	3.84	2.31	Premium	E	SI1 |
|2	3	0.23	2	2	5	56.9	65.0	327	4.05	4.07	2.31	Good	E	VS1 | 
|3	4	0.29	4	6	4	62.4	58.0	334	4.20	4.23	2.63	Premium	I	VS2 |
|4	5	0.31	2	7	2	63.3	58.0	335	4.34	4.35	2.75	Good	J	SI2 |

File: plotly/figure_factory/_facet_grid.py [hunk 2]
```diff
@@ -413,7 +413,7 @@ def _facet_grid_color_numerical(df, x, y, facet_row, facet_col, color_name,
                         mode='markers',
                         type=trace_type,
                         showlegend=False,
-                        **kwargs
+                        **kwargs_trace
```
Comment (Kully) [inline plotly/figure_factory/_facet_grid.py]: good catch

**Comment by Kully** [comment]
💃 from me!

**Comment by Kully** [comment]
💃

**Pull request closed without merging**